### ISSUE TITLE
`do` syntax can raise `no-unused-expressions`

### ISSUE THREAD

**Issue by jacobp100** [babel/babel-eslint#158]
Example’s worth a thousand words!

```js
const a = 1;
const b = do {
  a; // lint error here
}
```

**Comment by hzoo** [comment]
To solve this do we just visit all identifiers inside the do block?

**Comment by hzoo** [comment]
Actually maybe this should be a modification to the `no-unused-expressions` rule so it would probably be part of https://github.com/babel/eslint-plugin-babel. @jquense @mathieumg

**Comment by hzoo** [comment]
Closing in favor of moving this to https://github.com/babel/eslint-plugin-babel/issues/13
